js/coinsbit.js

```diff
@@ -0,0 +1,682 @@
+        }
+        params = this.omit (params, 'exchangeType');
+        return [exchangeType, params];
+    }
+
+    safeMarketObject (element) { // This method can be moved into base
+        // I see there are many new exchange classes (mexc, gateio, ...) where inside 'fetchMarkets' these properties are being set for spot-market objects (on each object) while in original exchange API responses those properties doesn't exist at all. So, these can be filled with this helper function (like 'safeTicker' does for each ticker object)
+        return this.deepExtend (element, {
+            'spot': undefined,
+            'derivative': undefined,
+            'contract': undefined,
+            'linear': undefined,
+            'inverse': undefined,
+            'contractSize': undefined,
+            'expiry': undefined,
+            'expiryDatetime': undefined,
+            'strike': undefined,
+            'optionType': undefined,
+            'settle': undefined,
+            'settleId': undefined,
+        });
+    }
+
+    validateTimestamp (methodName, ts) { // This method can be moved into base
+        // Validate if provided timestamp is in MILLISECONDS format (13 digits), and not SECONDS (10 digits)
+        if (ts && ts.toString ().length !== 13) {
+            throw new ExchangeError ('Provided timestamp to method "' + methodName + '" should be in Milliseconds (13 digit).');
+        }
+    }
+
+    async fetchStatus (params = {}) {
+        const [exchangeType, request] = this.checkExchangeType ('fetchStatus', params); // Here should be also exchangeType adpoted, because in my experience I've seen cases when spot is working, but futures/perp/etc is under maintenance
+        let response = '';
+        if (exchangeType === 'spot') {
+            response = await this.spotPublicGetSymbols (request); // This specific exchange (coinsbit) unfortunately, doesn't have any good way to get API status, so this is a specific case, I'm just making the most affordable endpoint call (better than nothing) to define the working-status of API. (However, overall structure might be like this)
+            //
+            // {"success":true,"message":"","result":["5BI_BUSD",...],"code":"200"}
+            //
+        } else if (exchangeType === 'futures') {
+            // JUST FOR DEMONSTRATIONAL PURPOSES
+            // response = await this.futuresPublicGetSymbols (params);
+        }
+        const code = this.safeString (response, 'code');
+        if (code !== undefined) {
+            const status = (code === '200') ? 'ok' : 'maintenance';
+            this.status = this.extend (this.status, {
+                'status': status,
+                'updated': this.milliseconds (),
+            });
+        }
+        return this.status;
+    }
+
+    async fetchTime (params = {}) {
+        return this.milliseconds ();
```

Comment:
We don't emulate this with the base call, and only implement it if they really have a true-endpoint for current time on the exchange-side time (not the client-side time).

Reply:
k